include/openthread-types.h

```diff
@@ -37,13 +37,19 @@
 
 #include <stdint.h>
 #include <stdbool.h>
+
 #include <platform/toolchain.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 /**
+ * This type represents the OpenThread instance structure.
+ */
+typedef struct otInstance otInstance;
```

Comment:
This doesn't require the caller to have any actual knowledge of the implementation. All my Windows driver code uses it just fine and has no internal file references.